server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardIntegrationTest.java

```diff
@@ -0,0 +1,384 @@
+
+        dataProbe.addUser(ALICE.asString(), PASSWORD);
+        dataProbe.addUser(BOB.asString(), PASSWORD);
+        dataProbe.addUser(CEDRIC.asString(), PASSWORD);
+        dataProbe.addUser(DELPHINE.asString(), PASSWORD);
+        dataProbe.addUser(FROM, PASSWORD);
+
+        mailRepositoryProbe = jamesServer.getProbe(MailRepositoryProbeImpl.class);
+    }
+
+    @AfterEach
+    void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Disabled("See JAMES-3944")
+    @Test
+    void forwardShouldRewriteTheSender() throws Exception {
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(BOB),
+                Mapping.forward(CEDRIC.asString()));
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(FROM, PASSWORD)
+            .sendMessage(FROM, BOB.asString());
+
+        Awaitility.await().until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1L);
+
+        SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+            MailKey mailKey = mailRepositoryProbe.listMailKeys(CUSTOM_REPOSITORY).get(0);
+            Mail mail = mailRepositoryProbe.getMail(CUSTOM_REPOSITORY, mailKey);
+
+            softly.assertThat(mail.getRecipients()).containsOnly(CEDRIC.asMailAddress());
+            softly.assertThat(mail.getMaybeSender().asOptional()).contains(BOB.asMailAddress());
+        }));
+    }
+
+    @Disabled("See JAMES-3944")
+    @Test
+    void latestForwardShouldRewriteTheSender() throws Exception {
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(BOB),
+                Mapping.forward(CEDRIC.asString()));
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(CEDRIC),
+                Mapping.forward(DELPHINE.asString()));
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(FROM, PASSWORD)
+            .sendMessage(FROM, BOB.asString());
+
+        Awaitility.await().until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1L);
+
+        SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+            MailKey mailKey = mailRepositoryProbe.listMailKeys(CUSTOM_REPOSITORY).get(0);
+            Mail mail = mailRepositoryProbe.getMail(CUSTOM_REPOSITORY, mailKey);
+
+            softly.assertThat(mail.getRecipients()).containsOnly(DELPHINE.asMailAddress());
+            softly.assertThat(mail.getMaybeSender().asOptional()).contains(CEDRIC.asMailAddress());
+        }));
+    }
+
+    @Disabled("See JAMES-3944")
+    @Test
+    void groupShouldBeAppliedToForwardedMails() throws Exception {
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(BOB),
+                Mapping.forward(DELPHINE.asString()));
```

Comment:
Shouldn't it be : 

```suggestion
                Mapping.forward(GROUP.asString()));
```
?